tests/tests/swfs/audio/aac/Test.as

```diff
@@ -0,0 +1,166 @@
+﻿package {
+	import flash.events.Event;
+	import flash.display.MovieClip;
+	import flash.display.Sprite;
+	import flash.media.Video;
+	import flash.net.URLRequest;
+	import flash.net.NetStream;
+	import flash.net.NetConnection;
+	import flash.utils.setTimeout;
+	import flash.media.SoundMixer;
+	import flash.utils.ByteArray;
+	import flash.geom.Point;
+
+
+	public class Test extends MovieClip {
+
+		public static function startTone(mc:MovieClip, file:String) {
+			trace("Starting tone:", file);
+
+			var connection:NetConnection = new NetConnection();
+			connection.connect(null);
+
+			var Client:Object = new Object();
+			Client.onMetaData = function(e) {};
+
+			var stream = new NetStream(connection);
+			stream.client = Client;
+			stream.play(file);
+
+			var video = new Video();
+			video.attachNetStream(stream);
+			mc.addChild(video);
+		}
+
+		public static function measureChannel(ba: ByteArray) : Point {
+			// Every tone has this frequency, regardless of channel count and sampling rate.
+			const FREQ = 444.0;
+
+			var sinSum = 0.0;
+			var cosSum = 0.0;
+			// The first multiplication by 2 is because of the buggy stretch factor
+			// (we only get every second sample).
+			var coeff = FREQ * 2 * Math.PI * 2 / 44100.0;
+
+			for (var j:uint=0; j<256; j++) {
+				var sp = ba.readFloat();
+				var arg = j * coeff;
+				sinSum += sp * Math.sin(arg);
+				cosSum += sp * Math.cos(arg);
+			}
+			sinSum /= 256.0;
+			cosSum /= 256.0;
+
+			// The returned complex number describes both the amplitude and the phase
+			// of the tone within the time window we happened to capture it.
+			return new Point(cosSum, sinSum);
+		}
+
+		public static function reconstructWave(phasor: Point) : ByteArray {
+			const FREQ = 444.0;
+			var ba = new ByteArray();
+			var coeff = FREQ * 2 * Math.PI * 2 / 44100.0;
+			for (var j:uint=0; j<256; j++) {
+				var arg = j * coeff;
+				var val = Math.cos(arg) * phasor.x + Math.sin(arg) * phasor.y;
+				// Need to compensate for the "every second sample only" bug here as well.
+				ba.writeFloat(val * 2.0);
+			}
+			ba.position = 0;
+
+			return ba;
+		}
+
+		public static function getFreq(ba: ByteArray) : Number {
```

Comment:
I guess this should have a better name.

Reply:
IMO it's fine as it's only a test :)

Reply:
Of course, but it's a remnant of a previous version where it actually still had a `freq` parameter, but now that it doesn't, I don't like it that much.

Reply:
Refactored the test a little bit... :)